File: judo-sdk/src/main/java/com/judopay/model/Receipts.java
package com.judopay.model;

import java.util.List;

/**
 * Represents the list of {@link Receipt} objects returned when querying the judo API for Receipts.
 */
@SuppressWarnings("unused")
public class Receipts {

    private final Integer resultCount;
    private final Integer pageSize;
    private final Integer offset;
    private final List<Receipt> results;

    public Receipts(final Integer resultCount, final Integer pageSize, final Integer offset, final List<Receipt> results) {
        this.resultCount = resultCount;
        this.pageSize = pageSize;
        this.offset = offset;
        this.results = results;
    }

    public Integer getResultCount() {
        return resultCount;
    }

    public Integer getPageSize() {
        return pageSize;
    }

    public Integer getOffset() {
        return offset;
    }

    public List<Receipt> getResults() {
        return results;
    }

    @Override
    public String toString() {
        return "Receipts{" +
                "results=" + results +
                ", offset=" + offset +
                ", pageSize=" + pageSize +
                ", resultCount=" + resultCount +
                '}';
    }

}

File: judo-sdk/src/main/java/com/judopay/api/model/response/Receipts.kt
package com.judopay.api.model.response

/**
 * Represents the list of [Receipt] objects returned when querying the judo API for Receipts.
 */
class Receipts(val resultCount: Int,
               val pageSize: Int,
               val offset: Int,
               val results: List<Receipt>) {

    override fun toString(): String {
        return "Receipts(resultCount=$resultCount, pageSize=$pageSize, offset=$offset, results=$results)"
    }
}